src/hardware/timer.cpp

```diff
@@ -221,7 +221,13 @@ static void write_latch(Bitu port,Bitu val,Bitu /*iolen*/) {
 		if (p->write_latch == 0) {
 			if (p->bcd == false) p->cntr = 0x10000;
 			else p->cntr=9999;
-		} else p->cntr = p->write_latch;
+		}
+		// square wave, count by 2
```

Comment:
Honestly, I don't understand the comment :)

Which one represents square wave and which one "count by 2" (but I still don't understand what it means - it means we check every second interrupt)? I presume `p->mode == 3` is square wave?

Do we want to check how "Co-authored-by:" trailer works in practice? You edited this fix already compared to what's in DOSBox-X, so perhaps a good idea would be to change authorship back to you and set jonathancampbell as Co-author. What do you think?

Reply:
Same.. I also don't really understand either comment, so retained them as-is.  I'd prefer not being a coauthor, as I really just dropped it in and slightly adjusted the formatting.

It would be nice clarify and expand on the comments though! @joncampbell123, can you enlighten us?

Reply:
I think all the explanation might be in https://github.com/joncampbell123/dosbox-x/issues/1492 - after reading it, perhaps we should include some information from there into the commit message.

Reply:
Mode 3 generates a square wave by counting down by 2. On counter overflow, the output is toggled. Counting by 2 allows the complete cycle of the square wave to fit in the countdown.

Reading back the counter in mode 3 should always show an even value for that reason. There is code in DOSBox SVN and all forks (DOSBox-X included) that masks the LSB and mentions that (and Corncob 3D) for that reason.

Take a look at the 8254 datasheet: http://hackipedia.org/browse.cgi/Computer/Platform/PC%2c%20IBM%20compatible/Timer/8253/8254%20programmable%20interval%20timer%20%281993%2d09%29%2epdf

Reply:
@joncampbell123 Thank you for the explanation, it's much clearer now :)

@kcgen Please amend the commit message and add this sentence in there. Keep the authorship and comment as it is, but add the following paragraph:

```
Mode 3 generates a square wave by counting down by 2. On counter
overflow, the output is toggled.  Counting by 2 allows the complete
cycle of the square wave to fit in the countdown.

See: https://github.com/joncampbell123/dosbox-x/issues/1492
```

And then signoff the commit: `git commit --amend --signoff` (in case I'll need to rebase it on the newer master which will overwrite the committer).